streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBGenericOptionsToDbOptionsColumnFamilyOptionsAdapter.java

```diff
@@ -304,7 +306,7 @@ public String walDir() {
 
     @Override
     public Options setWalDir(final String walDir) {
-        dbOptions.setWalDir(walDir);
+        LOGGER.warn("WAL is explicitly disabled by Streams in RocksDB. Setting option 'walDir' will be ignored");
```

Comment:
We could refactor out a helper for the log-and-return behavior with parameter as the name of `option`

Reply:
sounds like a good suggestion. I can extract the logging to a helper method. I think the return behaviour should still remain under the responsibility of the setters. Let me know what you think

Reply:
A logging helper method sounds reasonable. IMO the return can stay as it is, because we would not save too much. But I do not have strong feelings about it.